src/tools/debugger/session/Breakpoints.java

```diff
@@ -161,8 +169,14 @@ public boolean evaluate() {
 
  public synchronized BreakpointEnabling<MessageReceiverBreakpoint> getReceiverBreakpoint(
       final FullSourceCoordinate section) {
-    return receiverBreakpoints.computeIfAbsent(section,
+   BreakpointEnabling<MessageReceiverBreakpoint> bkp = receiverBreakpoints.computeIfAbsent(section,
         ss -> new BreakpointEnabling<>(new MessageReceiverBreakpoint(false, section)));
+
+   if (bkp.isDisabled() && stepping != null) {
+     bkp.setEnabled(stepping.getSteppingTypeOperation(section) == SteppingType.STEP_INTO);
```

Comment:
why is the enabling status of the breakpoint depending on some stepping info? and how can that stepping info be global? shouldn't it be local to an activity/actor?